drivers/serial/uart_nrfx_uarte.c

```diff
@@ -157,7 +173,23 @@ static inline NRF_UARTE_Type *get_uarte_instance(const struct device *dev)
 	return config->uarte_regs;
 }
 
-#ifdef UARTE_INTERRUPT_DRIVEN
+static void endtx_isr(const struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	int key = irq_lock();
+
+	if (nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ENDTX)) {
+		nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_ENDTX);
+		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STOPTX);
+	}
+
+	irq_unlock(key);
+
+}
+
+#if defined(UARTE_INTERRUPT_DRIVEN) || \
+	(!defined(UARTE_INTERRUPT_DRIVEN) && !defined(CONFIG_UART_ASYNC_API))
```

Comment:
This is wrong. For example, the following configuration will fail to build:
```
CONFIG_UART_INTERRUPT_DRIVEN=n
CONFIG_UART_ASYNC_API=y
CONFIG_UART_0_ASYNC=n
```

Reply:
please take a look now, i've added define which detects if any instance is enabled and not using async API and this is a condition on which interrupt handler is compiled
```
#if	(defined(CONFIG_UART_0_NRF_UARTE) && !defined(CONFIG_UART_0_ASYNC)) || \
	(defined(CONFIG_UART_1_NRF_UARTE) && !defined(CONFIG_UART_1_ASYNC)) || \
	(defined(CONFIG_UART_2_NRF_UARTE) && !defined(CONFIG_UART_2_ASYNC)) || \
	(defined(CONFIG_UART_3_NRF_UARTE) && !defined(CONFIG_UART_3_ASYNC))
#define UARTE_ANY_NONE_ASYNC 1
#endif
```